### PR TITLE
Fix array items to also handle raw types

### PR DESCRIPTION
**Pull request by dastbe**
- Add handling parser.rb to handle constructs like { "type" => "string" } in the items list/hash for an array
- Fix ObjC parser to account for this (and move some functions into Utils modules)
- Update tests for parser and Obj properties

### CODE REVIEW

**Comment by cknadler** [comment]
Dang. :+1:

I'll dig through this in a bit. Thanks for the PR!

File: test/nidyx/objc/test_property.rb [hunk 1]
```diff
@@ -34,14 +37,21 @@ def test_simple_array
...
 
   def test_typed_optional_array
-    obj = { "type" => ["array", "null"] }
+    obj = { TYPE_KEY => ["array", "null"] }
     p = property(obj, false)
     assert_equal(:array, p.type)
     assert_equal(["Optional"], p.protocols)
   end
 
+  def test_typed_array_with_protocls
```
Comment (sethfri) [inline test/nidyx/objc/test_property.rb]: Minor: Typo in "protocols"

**Comment by sethfri** [comment]
Ship It! Looks great to me. Chris, please give this a cursory glance and merge when you get some time.

**Comment by cknadler** [comment]
Looking.

File: lib/nidyx/parser.rb [hunk 2]
```diff
@@ -154,13 +155,24 @@ def resolve_array_refs(obj)
...
-        resolve_reference_string(item[REF_KEY])
-        types << class_name_from_ref(item[REF_KEY])
+      items.each do |i|
+        types << resolve_single_item(i)
       end
+
       types.compact
     end
 
+    # @param item (Hash) a single item
```
Comment (cknadler) [inline lib/nidyx/parser.rb]: TODO: formatting

File: lib/nidyx/objc/property.rb [hunk 3]
```diff
@@ -19,7 +21,8 @@ def initialize(property)
       @type_name = lookup_type_name(@type, property.class_name)
 
       @protocols = []
-      @protocols += property.collection_types if property.collection_types
+      # Exclude any primitive types for collections
+      @protocols += filter_primitives(property.collection_types) if property.collection_types
```
Comment (cknadler) [inline lib/nidyx/objc/property.rb]: TODO: write a test to ensure this is working.

File: lib/nidyx/parser.rb [hunk 4]
```diff
@@ -154,13 +155,24 @@ def resolve_array_refs(obj)
     # @return [Array] types contained in the array
     def resolve_items_array(items)
       types = []
-      items.each do |item|
-        resolve_reference_string(item[REF_KEY])
-        types << class_name_from_ref(item[REF_KEY])
+      items.each do |i|
```
Comment (cknadler) [inline lib/nidyx/parser.rb]: TODO: compact

File: lib/nidyx/objc/utils.rb [hunk 5]
```diff
@@ -0,0 +1,144 @@
+module Nidyx
+  module ObjCUtils
```
Comment (cknadler) [inline lib/nidyx/objc/utils.rb]: TODO: remove "module_function" and make calls explicit

File: lib/nidyx/objc/utils.rb [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+
+    SIMPLE_NUMBERS = Set.new [:unsigned, :integer, :number]
+
+    BOXABLE_NUMBERS = SIMPLE_NUMBERS + [:boolean]
+
+    FORBIDDEN_PROPERTY_PREFIXES = ["new", "copy"]
+
+    # @param type (Array[String|Symbol]) collection of types
+    # @return (Array[Symbol]) the collection w/out primitive types
+    def filter_primitives(ts)
```
Comment (cknadler) [inline lib/nidyx/objc/utils.rb]: TODO: Rename to something other than "primitives". Is more of "filter_standard_types" or "filter_inherent_types" as some of the types are objects, not just primitives.

**Comment by cknadler** [comment]
Looks great. Thanks for the changes David. Sorry for the wait. Merging now.

I'll follow up with the TODO items in separate issue (#38).

**Comment by cknadler** [comment]
Bugs are fixed. Waiting for CI to verify. Once thats done, I'll release this in 0.2.2.

**Comment by cknadler** [comment]
Released: https://rubygems.org/gems/nidyx/versions/0.2.2

**Pull request merged**